tests/wpt/mozilla/tests/mozilla/referrer-policy/generic/common.js

```diff
@@ -215,6 +241,56 @@ function queryScript(url, callback) {
   document.body.appendChild(script);
 }
 
+function queryCssLink(url, callback, referrer_policy) {
+  var x = document.createElement('script');
+  x.src = '/common/utils.js';
+  x.onerror = function() { console.log('whoops') };
+  x.onload = function() { doQuery() };
+  document.getElementsByTagName("head")[0].appendChild(x);
+
+  function doQuery() {
+    var id = token();
+    var link = document.createElement("link");
+
+    if (referrer_policy) {
+      for (var attr in referrer_policy) {
+        // TODO crashed when you assigned value to rel attribute
+        if (attr === "rel") {
+          link.relList.add("noreferrer");
+        } else {
+          link[attr] = referrer_policy[attr];
+        }
+      }
+    }
+
+    link.href = url + "&id=" + id;
+    link.relList.add("stylesheet");
+
+    // XXX Should be removed after "onload" event can be fired
+    var timeout_func = setTimeout(function timeout() {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', '/_mozilla/mozilla/referrer-policy/generic/subresource/stash.py?id=' + id, false);
+      xhr.onreadystatechange = function(e) {
+        if (this.readyState == 4 && this.status == 200) {
+          var server_data = JSON.parse(this.responseText);
+          // FIXME But why??
```

Comment:
The question here is why `server_data.headers` is str type only in QueryCssLink()? I assume it's due to my manually data posting in `generics/subresource/links.py`.

I'll try to figure this out after successfully calling stash putting method in `links.py` since it might be related.